Makefile

```diff
@@ -0,0 +1,51 @@
+HAS_DEP := $(shell command -v dep;)
+VERSION := $(shell sed -n -e 's/version:[ "]*\([^"]*\).*/\1/p' plugin.yaml)
+COMMIT_NO := $(shell git rev-parse HEAD 2> /dev/null || true)
+COMMIT := $(if $(shell git status --porcelain --untracked-files=no),"${COMMIT_NO}-dirty","${COMMIT_NO}")
+
+PKG:= github.com/openSUSE/helm-mirror
+LDFLAGS := -X $(PKG)/cmd.version=$(VERSION)
+LDFLAGS += -X $(PKG)/cmd.gitCommit=$(COMMIT)
+
+format:
+	test -z "$$(find . -path ./vendor -prune -type f -o -name '*.go' -exec gofmt -d {} + | tee /dev/stderr)" || \
+	test -z "$$(find . -path ./vendor -prune -type f -o -name '*.go' -exec gofmt -w {} + | tee /dev/stderr)"
+
+install: build
+	mkdir -p $(HELM_HOME)/plugins/helm-mirror/bin
+	cp bin/mirror $(HELM_HOME)/plugins/helm-mirror/bin
+	cp plugin.yaml $(HELM_HOME)/plugins/helm-mirror/
+
+build:
+	mkdir -p bin/
+	echo $(LDFLAGS)
+	go build -i -v -o bin/mirror -ldflags="$(LDFLAGS)"
+
+test:
+	rm -rf /tmp/mirror
+	go test -v ./... -cover -race
```

Comment:
As a tip, maybe it's a good time to introduce tools like `gofmt`, `vet` and `golint`. Take a look at [umoci](https://github.com/openSUSE/umoci/blob/master/Makefile#L92) for this. It's pretty easy to introduce now and I'd say that it's common practice among golang application.

Reply:
Thanks will check them out.